src/vocabularyChecker/detectWrongSyntax.ts

```diff
@@ -0,0 +1,142 @@
+import json from "./AllowedVocab.json";
+import Issue from "./Issue";
+const punctuation = new RegExp(/[\u2000-\u206F\u2E00-\u2E7F\\'!"#$%&()*+,\-.\/:;<=>?@\[\]^_`{|}~]/g);
+const space = new RegExp(/\s+/g);
+const allExceptSpace = new RegExp(/^(?!\s*$).+/g);
+const allowedWords = json;
+
+/**
+ * (input:) @param text
+ * This method takes the user input as a parameter.
+ * At first it prepocesses @param text by calling
+ * @function preprocessText thus calls the two
+ * functions @function eliminatePunctuation and @function splitIntoWords.
+ * For their functionality see the corresponding comments.
+ * @function checkInput then filters out all wrong @param word
+ * from @param text and puts them all into @param allWrongWords,
+ * which is printed in the console log by calling @function logInvalidWords.
+ * This is for debugging purposes and should later be changed to
+ * be displayed to the user. For further postprocessing all wrong
+ * words are collected in a @returns @param invalidWords of
+ * @type string array.
+ */
+
+export function checkInput(text: string) {
+		const preprocessedText: string[] = preprocessText(text);
+		const invalidWords = preprocessedText.filter(word => !allowedWords.includes(word));
+		// TODO: Make logInvalidWords dependent on this posOfInvalidWords.
+		const mapOfInvalidWords = createMapOfInvalidWords(text, invalidWords);
+		logInvalidWords(mapOfInvalidWords);
+		return invalidWords;
+}
+
+/**
+ * @param text
+ * @param invalidWords
+ * Returns a map of invalid words (consisted in @param invalidWords) and their corresponding
+ * positions (starting position and end position) inside of a string.
+ */
+
+export function createMapOfInvalidWords(text: string, invalidWords: string[]) {
+	const positions = new Map();
+	for (const word of invalidWords) {
+		positions.set(word, [text.indexOf(word), text.indexOf(word) + word.length - 1]);
```

Comment:
While tuple-types `[TYPE1, TYPE2, ... TYPEN]` are great to represent actual "pairs" of things and are sometimes necessary to use in certain APIs (like in `Map`s for their entries), they are not so good at establishing meaning and readability about their components. Especially when the components are of the same type, you gain not much from them.
Enter object-types (`{ prop1: TYPE1, prop2: TYPE2, ... propN: TYPEN }`). These allow you to give your components names and address them by that name, not needing to worry about their order at all. Example:

```typescript
function createPosition(startIndex: number, length: number): { fromIndex: number, toIndex: number } {
  return {
    fromIndex: startIndex,
    toIndex: startIndex + length,
  };
}
```

See how it's much more cleaner now to understand what the separate components mean?

If you plan on using an object type in multiple places, you can define it formally:

```typescript
type Position = { fromIndex: number, toIndex: number };
```

and you can even export those!

An alternative to defining object-types is defining a proper interface:

```typescript
interface Position {
  fromIndex: number;
  toIndex: number;
}
``` 

Since there is no downside to doing that, and interfaces can easily extend on one another, they are usually preferred. It's really the same thing though.

You *can* define structurally identical object-types or interfaces and assign them to another (Typescript uses a structural type system), but if something is used in multiple places and means the same everywhere, it should all refer to the same definition. In those cases, put the interface in its own file, export it from there and import it in all others.

Of course, you can also use full-size classes for the purpose of building data-structures, C#-style:

```
export class Position {
  public readonly fromIndex: number;
  public readonly toIndex: number;

  constructor(fromIndex: number, toIndex: number) {
    this.fromIndex = fromIndex;
    this.toIndex = toIndex; 
  }
}
```

These can also contain getters, setters, methods, static-functions ... it's usually overkill for something so simple though.

But it's all better than tuples.